### PR TITLE
refactor: reorganize tests, reduce repetition

### PR DESCRIPTION
**Pull request by sachinraja**
* test ESM build
* test commonjs `icons/*.js`
* extract icon tests to separate function to avoid code duplication

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -61,7 +61,7 @@
     "prepare": "is-ci || husky install",
     "prepublishOnly": "npm run build",
     "postpublish": "npm run clean",
-    "test": "jest",
+    "test": "node --experimental-vm-modules node_modules/jest/bin/jest.js",
```
Comment (sachinraja) [inline package.json:64]: Necessary for esm

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  testMatch: ['**/tests/**/*.test.?(m)js'],
+  moduleFileExtensions: ['js', 'mjs', 'json'],
```
Comment (sachinraja) [inline jest.config.js:3]: Necessary for esm

**Comment by mondeja** [comment]
Since #6505 has been merged you need to fix conflicts 🔧

**Comment by sachinraja** [comment]
Should be good now

File: package.json (hunk 1 above)
Comment (ericcornelissen) [reply package.json:64]: Alternatively we could [ask contributors](https://github.com/simple-icons/simple-icons/blob/develop/CONTRIBUTING.md#testing-package-locally) to use at least v14, but this is fine as well. Though I'd say it is ideally removed when LTS for v12 ends ([on 2022-04-30](https://nodejs.org/en/about/releases/))

File: tests/icons.test.mjs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  test(`${icon.title} has a valid svg value`, () => {
+    const svgFileContents = fs
+      .readFileSync(svgPath, 'utf8')
+      .replace(/\r?\n/, '');
+
+    expect(subject.svg.substring(subject.svg.indexOf('<title>'))).toEqual(
+      svgFileContents.substring(svgFileContents.indexOf('<title>')),
+    );
+  });
```
Comment (ericcornelissen) [inline tests/icons.test.mjs:29]: Shouldn't this be tested in `testIcon()`?

File: tests/test-icon.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+ * @param {String} slug Icon data slug
+ */
+const testIcon = (icon, subject, slug) => {
+  describe(icon.title, () => {
+    it('has the correct "title"', () => {
+      expect(subject.title).toStrictEqual(icon.title);
+    });
+
+    it('has the correct "slug"', () => {
+      expect(subject.slug).toStrictEqual(slug);
```
Comment (ericcornelissen) [inline tests/test-icon.js:14]: There might be a good reason for this, but why not:

```suggestion
      expect(subject.slug).toStrictEqual(icon.slug);
```

Comment (sachinraja) [reply tests/test-icon.js:14]: `icon.slug` won't be defined in all cases though. `icon` is directly from `_data/simple-icons.json`

File: package.json (hunk 1 above)
Comment (sachinraja) [reply package.json:64]: This is necessary even if you're on the latest Node version ([jest docs](https://jestjs.io/docs/ecmascript-modules)).

Comment (ericcornelissen) [reply package.json:64]: Thanks for pointing that out, didn't realize it's a Jest requirement :slightly_smiling_face:

File: tests/test-icon.js (hunk 4 above)
Comment (ericcornelissen) [reply tests/test-icon.js:21]: Ah indeed, my bad :sweat_smile: Thank you for pointing that out

(I knew there was a reason but couldn't think of it, felt it was better to comment than to leave it)

**Review by ericcornelissen — APPROVED** [review]

**Review by mondeja — APPROVED** [review]

**Pull request merged**